apis/_codegen/fetch-apis.go

```diff
@@ -1,99 +1,235 @@
 package main
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"go/format"
 	"io/ioutil"
-	"os"
+	"log"
+	"reflect"
+	"sort"
+	"sync"
 
 	got "github.com/taskcluster/go-got"
 	"github.com/taskcluster/taskcluster-cli/apis/definitions"
 )
 
 func main() {
+	// synchronization objects
+	mutex := &sync.Mutex{}
+	wg := &sync.WaitGroup{}
+
 	g := got.New()
 
+	gen := &generator{}
+
+	gen.Print("// Code generated by fetch-apis; DO NOT EDIT\n")
+	gen.Print("package apis\n")
+	gen.Print("\n")
+	gen.Print("import \"github.com/taskcluster/taskcluster-cli/apis/definitions\"\n")
+	gen.Print("\n")
+
 	// Fetch API manifest
 	res, err := g.Get("http://references.taskcluster.net/manifest.json").Send()
 	if err != nil {
-		fmt.Println("Failed to fetch api manifest, error: ", err)
-		os.Exit(1)
+		log.Fatalln("error: failed to fetch api manifest: ", err)
 	}
 	// Parse API manifest
 	var manifest map[string]string
 	if err = json.Unmarshal(res.Body, &manifest); err != nil {
-		fmt.Println("Failed to parse api manifest, error: ", err)
-		os.Exit(1)
+		log.Fatalln("error: failed to parse api manifest: ", err)
 	}
 
+	fmt.Println("Fetching Services:")
 	services := make(map[string]definitions.Service)
 	for name, referenceURL := range manifest {
-		fmt.Println(" - Fetching", name)
-		// Fetch reference
-		res, err2 := g.Get(referenceURL).Send()
-		if err2 != nil {
-			fmt.Println("Failed to fetch API: ", name, " error: ", err2)
-			os.Exit(1)
-		}
-		// Parse reference
-		var s definitions.Service
-		if err2 := json.Unmarshal(res.Body, &s); err2 != nil {
-			fmt.Println("Failed parse API: ", name, " error: ", err2)
-			os.Exit(1)
-		}
-		services[name] = s
+		wg.Add(1)
+		go func(n string, u string) {
+			s := fetchService(g, n, u)
```

Comment:
I did, I read through go-got and didn't see anything that would be thread-unsafe (no writes to shared objects/memory). I'll add a comment though.